cpp/src/utilities/device_atomics.cuh

```diff
@@ -125,34 +51,9 @@ namespace detail {
         return *( reinterpret_cast<T_output*>(&value) );
     }
 
-    // call atomic function with type cast between same underlying type
-    template <typename T, typename Functor>
-    __forceinline__  __device__
-    T typesAtomicOperation32(T* addr, T val, Functor atomicFunc)
-    {
-        using T_int = int;
-        T_int ret = atomicFunc(reinterpret_cast<T_int*>(addr),
-            cudf::detail::type_reinterpret<T_int, T>(val));
-
-        return cudf::detail::type_reinterpret<T, T_int>(ret);
-    }
-
-    // call atomic function with type cast between same underlying type
-    template <typename T, typename Functor>
-    __forceinline__  __device__
-    T typesAtomicOperation64(T* addr, T val, Functor atomicFunc)
-    {
-        using T_int = long long int;
-        T_int ret = atomicFunc(reinterpret_cast<T_int*>(addr),
-            cudf::detail::type_reinterpret<T_int, T>(val));
-
-        return cudf::detail::type_reinterpret<T, T_int>(ret);
-    }
-
-
     // -----------------------------------------------------------------------
     // the implementation of `genericAtomicOperation`
-    template <typename T, typename Op, size_t n>
+    template <typename T, typename Op, size_t N = sizeof(T)>
```

Comment:
Instead of having to explicitly pass in `sizeof(T)` as a template arg, how about just using `enable_if_t` on the `sizeof(T)`?

Reply:
It would be possible, but it may require a large part change...

Reply:
Sure, it's only a style issue anyways, so don't worry about it.